core/http/client/src/main/java/org/eclipse/rdf4j/http/client/QueueCursor.java

```diff
@@ -76,8 +75,21 @@ public void toss(Exception exception) {
 	public void put(E item)
 		throws InterruptedException
 	{
-		if (!done.get()) {
-			queue.put(item);
+		try {
+			while (!isClosed() && !done.get() && !Thread.currentThread().isInterrupted()
```

Comment:
After spending the last two weeks wrangling BlockingQueue uses I have really been turned off queue.put because of its blocking behaviour and the likelihood that thread interrupts are either silent because nothing was blocking when they occurred, or a thread somewhere silently swallowed them without resetting the interrupt status. I have changed this to be a 1-second wait loop, that will still respond during the 1 second offer immediately, but will check for closure/interrupts every 1 second before going back to the queue to offer the item again if it wasn't accepted the first time.

This is at the core of the stacktrace in #684, but on its own it was not going to ensure closure occurred, so the other changes are still necessary to get either the done or closed variables (and hopefully the thread interruption status) reliably propagated to the background thread operating here.